newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -137,221 +65,192 @@ type State = {|
+        return option.value;
+      } else {
+        return currentInputValue();
+      }
+    };
+
+    const handleChange = (event: Event, option: DataSource | string): void => {
+      if (option !== null) {
+        if (option.value) {
+          if (option.onClick) option.onClick();
+          else {
+            if (this.props.onChoose) this.props.onChoose(option.value);
+            else this.props.onChange(option.value);
+          }
+        } else {
+          /* option is just a string in this case, and it
+             enables user to type something other than provided options. */
+
+          this.props.onChange(option);
+        }
+        if (event.key) handleKeyDown(event.key);
+        if (this.props.onRequestClose) this.props.onRequestClose();
+      }
+    };
 
-                    // Events:
-                    InputProps: {
-                      // We wrap the onBlur/onFocus as we're a "semi controlled" field
-                      onBlur: wrappedOnBlur,
-                      onFocus: onFocus,
-                    },
+    //Todo: Find a way to call this function everytime.
+    const handleKeyDown = (key: string): void => {
+      if (key === 'Enter' && this._input.current !== null)
+        this._input.current.blur();
+    };
 
-                    // Props for the input field from downshift:
-                    inputProps,
+    const handleInputChange = (
+      e: Event,
+      value: string,
+      reason: string
+    ): void => {
+      this.setState({ inputValue: value });
+    };
+
+    const nullifyDefaultStyling = (params: Object): void => {
+      const { InputProps, inputProps, InputLabelProps } = params;
+      InputProps.className = null;
+      InputProps.endAdornment = null;
+      inputProps.className = null;
+      InputLabelProps.disableAnimation = true;
+      InputProps.style = styles.inputRoot;
+      inputProps.style = styles.inputInput;
+    };
+
+    const renderItem = (option: DataSource, state: Object): React.Node => {
+      if (option.type === 'separator') {
+        return (
+          <ListItem
+            divider
+            disableGutters
+            component={'div'}
+            style={{ padding: 0, margin: 0 }}
+          />
+        );
+      } else {
+        return (
+          <ListItem component={'div'} style={{ padding: 0, margin: 0 }}>
+            {option.renderIcon && (
+              <ListItemIcon>{option.renderIcon()}</ListItemIcon>
+            )}
+            {option.value}
+          </ListItem>
+        );
+      }
+    };
 
-                    // Style:
-                    style: props.textFieldStyle,
-                    fullWidth: props.fullWidth,
-                    ...computeTextFieldStyleProps(props),
+    const placeholder = () => {
+      if (this.props.errorText) return this.props.errorText.props.id;
+      else if (this.props.hintText && typeof this.props.hintText === 'string')
+        return this.props.hintText;
+      else return null;
```

Comment:
Never ever reading into this props of your props, at least in 99.9% of cases. By doing so, you're breaking the encapsulation of the children. You probably must use properly the i18n methods here - don't try to work around it :)